pages/pil/index.js

```diff
@@ -1,18 +1,49 @@
 const { Router } = require('express');
-const { buildModel } = require('../../lib/utils');
-const certificateSchema = require('./certificate/schema');
-const moduleSchema = require('./modules/schema');
-const proceduresSchema = require('./procedures/schema');
+const { cleanModel } = require('../../lib/utils');
+
+const createNewPilApplication = (req, res, next) => {
+  const opts = {
+    method: 'POST',
+    headers: { 'Content-type': 'application/json' }
+    // no body (we just want a blank pil returned with a new id)
+  };
+
+  req.api(`/establishment/${req.establishment}/profiles/${req.profile}/pil`, opts)
+    .then(({ json: { data } }) => {
+      return res.redirect(req.originalUrl.replace('create', data.id));
+    })
+    .catch(next);
+};
 
 module.exports = () => {
   const app = Router();
 
-  app.param('pil', (req, res, next, pil) => {
-    if (pil === 'create') {
-      req.model = buildModel(certificateSchema, moduleSchema, proceduresSchema);
-      req.model.id = 'new-training';
-      return next();
+  app.use('/', (req, res, next) => {
+    res.locals.static.establishment = req.user.profile.establishments.find(e => e.id === req.establishment);
+    res.locals.static.profile = req.model;
+    return next();
+  });
+
+  app.param('pil', (req, res, next, pilId) => {
+    if (pilId === 'create') {
+      console.log('attempting to create pil');
+      // only create a pil if we haven't already got one for this profile
+      if (!res.locals.static.profile.pil) {
```

Comment:
This seems a fragile way of checking for a pil, can't we use req.profile or something?